[SUREFIRE-1453] Allow to specify non existant classes as groups

eolivelli opened the pull request:
Allow to specify class names which does not resolve to a class in the scope of the project classpath.
This will be useful for instance in reactor projects in which non all modules contain the requested marker interface in the classpath.
See SUREFIRE-1453 for a complete description

Tibor17 commented:
@eolivelli 
Thx for your work. Try to prove it with integration test. We always have to do it if it technically can be tested.

Tibor17 commented:
Try to name the commit with brackets:
` [SUREFIRE-1453] Allow to specify non existant classes as groups`

eolivelli commented:
@Tibor17 I have added some IT.
I found a bug in JUnit48TestCategoriesIT, actually testCategoriesABForkAlways did not call ".forkAlways()"

I have added an IT test for the case of no class matching the filter.
I have enhanced the testCategoriesAC in a way that it tests only using Simple Class Names and not Fully Qualified  Class Names.
In fact I saw that the original idea in was to let the use pass even suffixes:
`if ( name.endsWith( enabled ) )`
But actually could not work because the 'pattern' had to be a resolvable class name.

I did changed names in parameters in JUnit48TestCategoriesIT to make them more "consistent"

Tibor17 commented:
@eolivelli 
Thx for contributing. The PR was pushed to master and now it can be closed.

eolivelli commented:
You are welcome!

jira-importer commented:
Resolve #2193
